src/App.php

```diff
@@ -78,17 +78,19 @@ private function getEventManager()
     /**
      * Application execution.
      *
-     * @param mixed|null $request  Representation of an outgoing, client-side request.
-     * @param mixed|null $response Representation of an incoming, server-side response.
+     * @param mixed|null $request  Representation of an outgoing,
+     *  client-side request.
+     * @param mixed|null $response Representation of an incoming,
+     *  server-side response.
      *
      * @return mixed
      */
     public function run($request = null, $response = null)
     {
-        $request = $request ?: ServerRequestFactory::fromGlobals();
-        $response = $response ?: new Response();
+        $request = $request ?: $this->getContainer()->get("request");
+        $response = $response ?: $this->getContainer()->get("response");
```

Comment:
getContainer is a private API :)

2015-10-08 10:57 GMT+02:00 Abdul Malik Ikhsan <notifications@github.com>:

> In src/App.php
> <https://github.com/pennyphp/penny/pull/88#discussion_r41490323>:
>
> >       *
> >       * @return mixed
> >       */
> >      public function run($request = null, $response = null)
> >      {
> > -        $request = $request ?: ServerRequestFactory::fromGlobals();
> > -        $response = $response ?: new Response();
> > +        $request = $request ?: $this->getContainer()->get("request");
> > +        $response = $response ?: $this->getContainer()->get("response");
>
> use $this->container instead of $this->getContainer() for internally
> call, as I cleaned up on previous PR
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pennyphp/penny/pull/88/files#r41490323>.
>



-- 
Gianluca Arbezzano
www.gianarb.it